core/src/main/scala/cats/InvariantChoosable.scala

```diff
@@ -0,0 +1,62 @@
+package cats
+
+import simulacrum.typeclass
+import cats.data.INothing
+
+/**
+ * An Invariant Additive Monoidal.
+ * Adds the ability to create an `empty` value which must serve as the identity for `choice`.
+ *
+ * Must obey the laws defined in cats.laws.InvariantChoosableLaws.
+ */
+@typeclass trait InvariantChoosable[F[_]] extends InvariantChoice[F] {
+  def zero: F[INothing]
```

Comment:
This only works in the covariant case. If we have an `F[Nothing]`, we can't get an `F[A]` using `contramap` or `imap` as it would mean needing a function that returns `Nothing`.